docker/build/devdeps.Dockerfile

```diff
@@ -170,7 +170,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         git ninja-build libcurl4-openssl-dev \
         python3 python3-pip libpython3-dev \
     && python3 -m pip install --no-cache-dir \
-        lit pytest numpy \
+        pybind11==2.10 lit pytest numpy \
```

Comment:
I prefer we remove pybind from tpls, and replace with an actual git submodule pointing to the 2.10 release branch of pybind. Since pybind is a purely build-time dependency (not needed for the deployment of the resultant wheel), and header-only (no libraries to link), having it as a submodule makes life easier for developers and people building from source (no need to make sure it is installed on your system, git clone cuda quantum and you've got it).

Reply:
Looks like we need it as a submodule in any case to do some minimal customization.